src/lib/transaction_snark_scan_state/transaction_snark_scan_state.ml

```diff
@@ -1350,10 +1350,19 @@ let fill_work_and_enqueue_transactions t transactions work =
     incomplete_txns_from_recent_proof_tree t
   in
   let%bind work_list = fill_in_transaction_snark_work t.scan_state work in
+  let added_works = List.length work in
+  let merge_jobs_created = List.length work_list in
```

Comment:
Shouldn't we just compute that within the metadata structure since it's used only for this purpose?

Reply:
True, I will change that.